siobrultech_protocols/gem/api.py

```diff
@@ -92,9 +39,22 @@ async def send(arg: T) -> R:
         protocol.end_api_request()
 
 
+class GEMAPIStringResponseParser(Generic[R]):
+    """Most API responses in the GEM API end with \r\n; this parser waits until it sees a \r\n before attempting to parse."""
+
```

Comment:
"Most", but if it's an API that doesn't, how should they handle it?  Can we include some docs?

Reply:
Are the docs for `ApiCall` enough?

```
    def __init__(
        self, formatter: Callable[[T], str], parser: Callable[[str], R | None]
    ) -> None:
        """
        Create a new APICall.

        formatter - a callable that, given a parameter of type T, returns the string to send to the GEM to make the API call
        parser - a callable that, given a string, parses it into a value of type R.
                If there is not enough data to parse yet, it should return None.
                If there is enough data to parse, but it is malformed, it should raise an Exception."""
```